tools/tsp-client/package.json

```diff
@@ -38,7 +38,8 @@
     "@types/mocha": "^10.0.1",
     "@types/node": "^20.4.8",
     "@types/prompt-sync": "^4.2.1",
-    "@typespec/compiler": "0.50.0",
+    "@typespec/compiler": "^0.58.0",
+    "@azure-tools/typespec-autorest": "^0.44.0", 
```

Comment:
Do folks need to use v0.44.0 or later to be able to run this command?

Reply:
sorting API is introduced recently into autorest.

Reply:
Makes sense, then we should also have it under dependencies.

Reply:
Ah yeah should be in dependencies not devDependencies

Reply:
Also, I think we should float peerDependencies for typespec/compiler between 0.58.0 - 1.0.0

Reply:
^0.44.0 is `0.44.0 <= ver < 1.0.0`

Reply:
no for `0.` `^` doesn't allow next minors `^0.44.0` its just `>=0.44.0 && < 0.45.0`.

However in this case its not meant to be used as a peerDependency but as a regular dependency so this is correct.
This dependency can be updated whenever(probably will complain about the version of the compiler if yhou don't update) but in theory it doesn't matter its just loading the function from this lib

Reply:
Also, the comment was pointing out that these deps should be under "dependency"

Reply:
We float typespec/compiler in peerDependencies, which is where I was saying we need to update it to 0.58.0

Reply:
I'll fix the peer deps in a separate pr @allenjzhang. Resolving now

Reply:
https://github.com/Azure/azure-sdk-tools/issues/8766 tracking issue before next tsp-client release